features/dd-sdk-android-trace/src/main/java/com/datadog/opentracing/propagation/DatadogHttpCodec.java

```diff
@@ -6,116 +6,167 @@
+                String origin = null;
+                String mostSignificant64BitsTraceIdAsHex = null;
+                String leastSignificant64BitsTraceIdAsDecimal = null;
+
+
+                for (final Map.Entry<String, String> entry : carrier) {
+                    final String key = entry.getKey().toLowerCase(Locale.US);
+                    final String value = entry.getValue();
+
+                    if (value == null) {
+                        continue;
+                    }
+
+                    if (LEAST_SIGNIFICANT_TRACE_ID_KEY.equalsIgnoreCase(key)) {
+                        leastSignificant64BitsTraceIdAsDecimal = value;
+                    } else if (DATADOG_TAGS_KEY.equalsIgnoreCase(key)) {
+                        mostSignificant64BitsTraceIdAsHex = extractMostSignificant64BitsTraceId(value);
+                    } else if (SPAN_ID_KEY.equalsIgnoreCase(key)) {
+                        spanId = validateUInt64BitsID(value, 10);
+                    } else if (SAMPLING_PRIORITY_KEY.equalsIgnoreCase(key)) {
+                        samplingPriority = Integer.parseInt(value);
+                    } else if (ORIGIN_KEY.equalsIgnoreCase(key)) {
+                        origin = value;
+                    } else if (key.startsWith(OT_BAGGAGE_PREFIX)) {
+                        if (baggage.isEmpty()) {
+                            baggage = new HashMap<>();
+                        }
+                        baggage.put(key.replace(OT_BAGGAGE_PREFIX, ""), HttpCodec.decode(value));
+                    }
+
+                    if (taggedHeaders.containsKey(key)) {
+                        if (tags.isEmpty()) {
+                            tags = new HashMap<>();
+                        }
+                        tags.put(taggedHeaders.get(key), HttpCodec.decode(value));
+                    }
+                }
+                if(leastSignificant64BitsTraceIdAsDecimal==null|| mostSignificant64BitsTraceIdAsHex==null){
+                    return new TagContext(origin, tags);
+                }
+
+                final long leastSignificantTraceId =
+                        LongStringUtils.parseUnsignedLong(leastSignificant64BitsTraceIdAsDecimal);
+                final String traceIdAsHex = mostSignificant64BitsTraceIdAsHex +
+                        LongStringUtils.toHexStringPadded(leastSignificantTraceId, 16);
+                final BigInteger traceId = validateUInt128BitsID(traceIdAsHex, 16);
+                if (!BigInteger.ZERO.equals(traceId)) {
+                    final ExtractedContext context =
+                            new ExtractedContext(traceId, spanId, samplingPriority, origin, baggage, tags);
+                    context.lockSamplingPriority();
+
+                    return context;
+                } else if (origin != null || !tags.isEmpty()) {
+                    return new TagContext(origin, tags);
+                }
+            } catch (final RuntimeException e) {
+            }
```

Comment:
do we not want to log/handle this?

Reply:
this is old imported java code and the intention was just to add the 128 bits support here and not to refactor this